### ISSUE TITLE
Discussion: Is it me or the configuration options could be more minimalistic

### ISSUE THREAD

**Issue by doronbehar** [beetbox/beets#2627]
Excuse me for not following exactly the Issue template, but I hope it's OK since It's just a question I'd would like to open for discussion:
I'm rather confused with some of the configuration options, especially in the `import` section - Why is there both the options `import.copy` and `import.move`? Wouldn't it be simpler to use one option only? And here's another example - According [this page in the documentation](http://beets.readthedocs.io/en/latest/reference/config.html#link), It seems that `import.link` is boolean, OK but it conflicts with `import.hardlink`. Would it be possible to have only `import.link` which could accept the value `hard` as well?

According to the documentation, it seems that `import.copy` and `import.move` may conflict and `import.link` and `import.hardlink` may conflict as well between them and with the previous options as well! I think it will be wiser to have 1 option named something like: `import.operation` and it's optional values should be `copy`, `move`, `link` and `hardlink`.

What do you think? Is it worth making a PR for this?

**Comment by sampsyo** [comment]
Yes! All of these options are mutually exclusive, so one big option like `file_action` or something could make sense. The only reason it is the way it is now is because each option was added over time—only in retrospect does it looks like way too many options.

We could explore switching over to one big, global option, but here are a few things to consider:

- One other behavior—not doing anything at all, leaving files in place—should be possible.
- The command-line switches would also need to be addressed, although it may be possible to have them work the same way.
- Backwards-compatibility would be nice, and fortunately doesn't seem that hard to achieve while making the switch.

**Comment by doronbehar** [comment]
I totally agree with you.
As for backwards compatibility: What if we will declare the old options as deprecated and the next release will deliver a red message of deprecation in any operation of `beet`? This way people will get the time to change their configuration file if they will run the program at least once.

**Comment by sampsyo** [comment]
We’ve deprecated config options in the past, and a lackadaisical approach seems to work well: we add support for the new option, print a deprecation message when people use the old one (but still support it), and leave both options in place indefinitely until it seems like a suitably long period of time has passed. This usually incurs little maintenance overhead and lets users move at their own pace.

**Comment by doronbehar** [comment]
I think you have made a good point.
I don't have enough experience with python, yet I have made a PR in the past similar in difficulty and it wasn't that hard. Could you assign me to work on this issue? This way it will show up in my [TaskWarrior](http://taskwarrior.org/) list with [bugwarrior](https://github.com/ralphbean/bugwarrior).
